internal/secrets/secret_test.go

```diff
@@ -0,0 +1,186 @@
+package secrets
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/buildkite/agent/v3/api"
+)
+
+// mockAPIClient implements the APIClient interface for testing
+type mockAPIClient struct {
+	secrets map[string]*api.Secret // map of secret key to secret response
+	errors  map[string]error       // map of secret key to error to return
+}
```

Comment:
instead of mocking the API client, i'd recommend using `httptest` to mock out a server for a real API client to talk to, as it puts more of our production code under test.  [`agent/agent_worker_test.go`](https://github.com/buildkite/agent/blob/main/agent/agent_worker_test.go#L37-L80) is some good prior art for this

Reply:
Actioned